operator/main.go

```diff
@@ -177,34 +202,20 @@ func runOperator() {
 		operatorMetrics.Register()
 	}
 
-	k8s.Configure(
-		option.Config.K8sAPIServer,
-		option.Config.K8sKubeConfigPath,
-		float32(option.Config.K8sClientQPSLimit),
-		option.Config.K8sClientBurst,
-	)
-	if err := k8s.Init(option.Config); err != nil {
-		log.WithError(err).Fatal("Unable to connect to Kubernetes apiserver")
-	}
-	close(k8sInitDone)
+	initK8s(k8sInitDone)
 
-	// We try to update the Kubernetes version and capabilities. If this fails,
-	// we cannot move forward as we can misinterpret the available Capabilities.
-	// For example, in a case where we got an error when checking for Leases support
-	// we should not assume that support is not available and run the operator
-	// in non HA mode. As this can conflict with other operator instances running
-	// in the cluster, that figured the capabilities correctly and is running in HA
-	// mode.
-	if err := k8sversion.Update(k8s.Client(), option.Config); err != nil {
-		log.WithError(err).Fatal("Unable to update Kubernetes capabilities")
-	}
 	capabilities := k8sversion.Capabilities()
-
 	if !capabilities.MinimalVersionMet {
 		log.Fatalf("Minimal kubernetes version not met: %s < %s",
 			k8sversion.Version(), k8sversion.MinimalVersionConstraint)
 	}
 
+	// Register the CRDs after validating that we are running on a supported
+	// version of K8s.
+	if err := k8s.RegisterCRDs(); err != nil {
+		log.WithError(err).Fatal("Unable to register CRDs")
+	}
```

Comment:
One question I have is: should this be done only by the leader in case of Operation HA? Or is it safe for multiple Operators to try? 

My impression that it's likely OK because we will only modify the schema version to the latest, so if there was ever a conflicting change, the latest version will win out.

Reply:
Yes, to install the CRD faster it can be done outside of the leader election.